sphinx/source/docs/dev_guide/creating_pull_requests.rst

```diff
@@ -0,0 +1,153 @@
+.. _devguide_pull_requests:
+
+Creating a successful pull request
+==================================
+
+All changes to Bokeh's codebase and documentation are managed through
+`pull requests`_.
+
+Prerequisites
+-------------
+
+Creating a pull request requires some basic knowledge of GitHub. See the
+`GitHub documentation`_ for general information about GitHub and pull requests.
+
+To create a pull request, you need a development environment as described in
+:ref:`devguide_setup`.
+
+You can format any text you enter when working with pull requests. To learn more
+about your formatting options, see `Writing on GitHub`_ in the
+`GitHub documentation`.
+
+Pull request workflow
+---------------------
+
+1. Find an issue to reference
+    Every pull request to the Bokeh repository begins with an issue. See
+    `Pull Requests in the Bokeh wiki <wiki pull requests_>`_ for more
+    information on exceptions.
+
+    First, pick an issue from Bokeh's `issue tracker`_. Issues that are
+    relatively easy to get started on are labeled `good first issue`_.
+
+    Next, post a comment in the issue's discussion to indicate that you are
+    working on this issue. If there is no issue yet, you should first
+    :ref:`create one <devguide_issues>`.
+
+2. Create a branch
+    Before writing any code, you need to create a new `branch`_ on
+    :ref:`your fork <devguide_setup_cloning>` of the Bokeh repository. In most
+    cases, you should base your new branch on Bokeh's ``default`` branch. The
+    default branch is usually a branch that represents the next version of
+    Bokeh, for example ``branch-2.4``.
+
+    Use this pattern to name your branch:
+
+    ``[issue number]_[Python/BokehJS model you are working on, if applicable]_[optional: model property or short description]``
+
+    For example:
+
+    ``11423_table_column_visible``
+
+    See `BEP 6: Branching Strategy`_ for more information on the different
+    branches in the Bokeh repository.
+
+3. Publish your branch and push a first commit
+    Before you can open a pull request, you first need to publish your branch.
+    At this time, you should also include a first commit.
```

Comment:
I haven't really heard "publish" in this context. I think usually just more like "push the branch from your local clone, to your fork on GitHub" and it would be good to link to GH docs or tutorials about that I think (or an example command)